app/controllers/api/v1/statuses_controller.rb

```diff
@@ -47,6 +48,22 @@ def context
       ancestors_limit         = ANCESTORS_LIMIT
       descendants_limit       = DESCENDANTS_LIMIT
       descendants_depth_limit = DESCENDANTS_DEPTH_LIMIT
+    else
+      unless @status.local? && !@status.should_fetch_replies?
+        json_status = fetch_resource(@status.uri, true, current_account)
```

Comment:
This will take a while and possibly fail due to networking issues. What if you extracted this section to a worker, and returned HTTP 206 Partial Response to signal to the client that there might be more replies coming in a future response?

Reply:
I was thinking of adding the reply URI to the status model to avoid the network request in the context endpoint, because we have this at the time any status is created via activitypub (and should be able to fill it in for local posts without too much difficulty). I didn't mostly just to get it working but also i figured that adding new columns would be discouraged, but it makes sense to have since this at least doubles the number of requests per status we have to do.

thx for the tip on 206, i didn't even think of that

Reply:
Saving the replies collection URI kind of makes sense but it's also going to add gigabytes of data into the `statuses` table (on mastodon.social this table has roughly 701977540 rows, an example replies URI has 73 characters, I think that's roughly 51 GB of data if all the rows had it from the start?) so if it can be avoided maybe it should.

More on HTTP 206, we do a similar trick when uploading media. The `GET /api/v1/media/:id` endpoint returns HTTP 206 while the media is processing; the client keeps polling it until it gets HTTP 200, this is how the "Processing..." progress bar state is implemented in the composer.

Reply:
Would it help the case for storing the replies uri if we stored it as a relative uri vs. The status urI? We could hide that behind a getter/setter so it always appeared as the full URL everywhere else but on disk we could store it cheaply?

There are probably only a few dozen unique relative paths from a status to its replies (I would bet ~hundreds, realistically, given whacky single-user homebrew impls), not sure if the db compresses at all or if there's a cheap way to store discrete values as an int:string map in postgres, but that would probably be orders of magnitude cheaper